File: Blind 75 Leetcode problems/Arrays/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int n=nums.size();
        int maxsum=nums[0];
        int currentsum=nums[0];
        for (int i=1 ; i<n ; i++){
            currentsum=max(nums[i],currentsum+nums[i]);
            maxsum=max(maxsum,currentsum);
            }
        return maxsum;
    }
};

File: maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int currsum=0;
        int maxsum=nums[0];
        for(int i=0;i<nums.length;i++){
            currsum+=nums[i];
            if(currsum>maxsum) maxsum=currsum;
            if(currsum<0) currsum=0;
        }
    
        return maxsum;
    }
}
